snuba/query/logical.py

```diff
@@ -217,6 +224,9 @@ def get_conditions(self) -> Optional[Sequence[Condition]]:
     def get_condition_from_ast(self) -> Optional[Expression]:
         return self.__condition
 
+    def replace_ast_condition(self, condition: Expression) -> None:
+        self.__condition = condition
```

Comment:
The rest of this class uses `set_` prefix instead of `replace_` - should we keep things consistent with that?

Reply:
The `set_` are from the old representation. I used replace for the AST to remind that, this class, when you get it, is already initialized by the parser, so we would have two set of writer methods: those that add stuff to a clause (like add_condition_to_ast) and those that wipe it and replace with something else, which should be used with care.

That's the only reason, I am ok moving back to `set` if it feels more clear.

Reply:
It seems `set` and `replace` should be used with equal care and are doing similar types of operations (distinct from `add`) so its nice to keep a same prefix. I assume that least some of the `set_` methods won't be going away even if most will (e.g. `set_data_source`) so maybe still good to align with that?

Reply:
Replaced with setters.